gcp/apis/compute/v1alpha1/network_types.go

```diff
@@ -0,0 +1,398 @@
+
+// GetClassReference of this ReplicationGroup.
+func (n *Network) GetClassReference() *corev1.ObjectReference {
+	return n.Spec.ClassReference
+}
+
+// SetWriteConnectionSecretToReference of this ReplicationGroup.
+func (n *Network) SetWriteConnectionSecretToReference(r corev1.LocalObjectReference) {
+	n.Spec.WriteConnectionSecretToReference = r
+}
+
+// GetWriteConnectionSecretToReference of this ReplicationGroup.
+func (n *Network) GetWriteConnectionSecretToReference() corev1.LocalObjectReference {
+	return n.Spec.WriteConnectionSecretToReference
+}
+
+// GetReclaimPolicy of this ReplicationGroup.
+func (n *Network) GetReclaimPolicy() v1alpha1.ReclaimPolicy {
+	return n.Spec.ReclaimPolicy
+}
+
+// SetReclaimPolicy of this ReplicationGroup.
+func (n *Network) SetReclaimPolicy(p v1alpha1.ReclaimPolicy) {
+	n.Spec.ReclaimPolicy = p
+}
+
+// +kubebuilder:object:root=true
+
+// NetworkList contains a list of Network
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Network `json:"items"`
+}
+
+// GCPNetworkSpec contains fields of googlecompute.Network object that are
+// configurable by the user, i.e. the ones that are not marked as [Output Only]
+// In the future, this can be generated automatically.
+type GCPNetworkSpec struct {
+	// IPv4Range: Deprecated in favor of subnet mode networks. The range of
+	// internal addresses that are legal on this network. This range is a
+	// CIDR specification, for example: 192.168.0.0/16. Provided by the
+	// client when the network is created.
+	IPv4Range string `json:"IPv4Range,omitempty"`
+
+	// AutoCreateSubnetworks: When set to true, the VPC network is created
+	// in "auto" mode. When set to false, the VPC network is created in
+	// "custom" mode.
+	//
+	// An auto mode VPC network starts with one subnet per region. Each
+	// subnet has a predetermined range as described in Auto mode VPC
+	// network IP ranges.
+	AutoCreateSubnetworks *bool `json:"autoCreateSubnetworks,omitempty"`
+
+	// Description: An optional description of this resource. Provide this
+	// field when you create the resource.
+	Description string `json:"description,omitempty"`
+
+	// Name: Name of the resource. Provided by the client when the resource
+	// is created. The name must be 1-63 characters long, and comply with
+	// RFC1035. Specifically, the name must be 1-63 characters long and
+	// match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?. The first
+	// character must be a lowercase letter, and all following characters
+	// (except for the last character) must be a dash, lowercase letter, or
+	// digit. The last character must be a lowercase letter or digit.
+	Name string `json:"name,omitempty"`
+
+	// RoutingConfig: The network-level routing configuration for this
+	// network. Used by Cloud Router to determine what type of network-wide
+	// routing behavior to enforce.
+	RoutingConfig *GCPNetworkRoutingConfig `json:"routingConfig,omitempty"`
```

Comment:
To be honest, I didn't give much thought to this because the initial intention was to have exact mirror of corresponding `googlecompute.Network` object and then aim for the least divergence from that object. I think we need to discuss our approach on this in the context of how code generation would look like, do we still want it (can we afford having to bend things if needed because of provider's api)?

Reply:
I have opened #741 to track this.